### PR TITLE
Add a new blog post about block device compatibility feature in go-pmem

### PR DESCRIPTION
**Pull request by jerrinsg**

### CODE REVIEW

File: _posts/2020-02-06-Block-Device-Compatibility.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+The block device compatibility layer of *go-pmem* was developed so that
+our package can be used even on those machines without persistent memory (pmem).
+
+This layer provides weaker consistency guarantee than *go-pmem* running on
+persistent memory, but lets developers use and test our package on machines
+which has only block I/O storage. The weaker consistency guarantees offered by
+this compatibility layer are the following:
+* Our package is resilient to application crashes or restarts
+* In case of a host crash or restart, data can get irrecoverably corrupted
```
Comment (pratapsu) [inline _posts/2020-02-06-Block-Device-Compatibility.md]: rewrite as:

...are the following:
 * it is resilient to application crashes or restarts
 * ...

File: _posts/2020-02-06-Block-Device-Compatibility.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+## Innards
+
+To ensure writes to a pmem device is made persistent, *go-pmem* issues cache
+flushing and memory barrier instructions as required. But if the file passed to
+```pmem.Init()``` function resides in a block I/O device, then an ```msync()```
+system call has to be issued to ensure data writes are persistent. ```msync()```
+flushes to disk data at a page granularity. Therefore, issuing ```msync()``` for
+byte-level data updates leads to write-amplification and significantly slowdown
```
Comment (pratapsu) [inline _posts/2020-02-06-Block-Device-Compatibility.md]: significantly --> significant.

File: _posts/2020-02-06-Block-Device-Compatibility.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+of applications. To offer a reduced consistency guarantee to applications,
+but at the same time let them use our package on a machine without pmem, we
+developed a compatibility layer to *go-pmem*. If *go-pmem* identifies that the
+pmem file resides on a block I/O device, no explicit ```msync()``` is issued to
+ensure data persistence. Instead, we rely on the kernel to periodically flush
+dirty data from the DRAM to disk. Dirty data is also flushed to persistent media
```
Comment (pratapsu) [inline _posts/2020-02-06-Block-Device-Compatibility.md]: is also flushed to -> is also flushed, by the operating system, to

**Review by pratapsu — COMMENTED** [review]
You need to have a conclusion!

**Comment by jerrinsg** [comment]
Thanks for the review @pratapsu . I have incorporated your feedback.

**Pull request merged**